### PR TITLE
Adding in new logging and error handling guide. 

### PR DESCRIPTION
**Pull request by jwoertink**
Reordered a few pages. I wasn't sure the best order, so I can put them in another order if needed.

### CODE REVIEW

**Comment by jwoertink** [comment]
@paulcsmith 
> failed during stage 'building site': Build script returned non-zero exit code: 42

Any idea what that means? Maybe I missed something?

**Comment by paulcsmith** [comment]
I needed to update the gems. Could you rebase and force push? Then it should be able to deploy the preview ok

File: source/guides/logging-and-error-handling.html.md.erb [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+# Returns the status_code colorized with green, yellow, or red text based on the range of the code
+colored_status_code(context.response.status_code)
+
+# Formats the time to ISO_8601_DATE_TIME if the `show_timestamps` option is set to true.
+timestamp(time)
+
+# Does some fancy math to display the elapsed span in a human readable format
+elapsed_text(elapsed)
+```
+
```
Comment (paulcsmith) [inline source/guides/logging-and-error-handling.html.md.erb:79]: Very nice writeup! 👍

File: source/guides/logging-and-error-handling.html.md.erb [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+There's many different services out there where you can ship your exceptions off to for better cataloging and searching of the errors.
+
+In your `src/actions/errors/show.cr` file, there are several different `handle_error` methods. When an exception occurs in one of your actions, the appropriate method is called passing in the exception. This gives you a chance to grab that and pass it on. You can also create your own override if you want to report specifically on a custom error.
+
+```crystal
+def handle_error(error : MyCustomError)
+  ErrorReporter.report(context, error)
+  render Errors::ShowPage, status: 500, title: "Oops!"
+end
+```
```
Comment (paulcsmith) [inline source/guides/logging-and-error-handling.html.md.erb:152]: I think that in general using `handle_error `in `Errors::Show` should be preferred over including `Lucky::HttpRespondable`. The reason is that you get a lot more flexibility in the `Errors::Show`. You can control what the body should be, you can change it based on content-type, etc. 

So could you show the example exception class and then show an example `Errors::Show#handle_error` method? Then maybe keep the part you have now below as another example. I think it is still useful, but is usually more useful for libraries. What do you think? Also I think the order is great! We can tweak it later if necessary

File: source/guides/logging-and-error-handling.html.md.erb [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+
+## Logger Options
+
+### show_timestamps : Bool
+Show/Hide the current timestamp of the request. By default this is shown only in production
+
+### log_formatter : Lucky::LogFormatters::Base
+Customize how your log is formatted, and what information it shows. By default this is the `DefaultLogFormatter`
+
+### enabled : Bool
```
Comment (paulcsmith) [inline source/guides/logging-and-error-handling.html.md.erb:29]: Could you add newlines underneath these `###`

**Review by paulcsmith — COMMENTED** [review]
This is awesome! Can't wait to get this in!

File: source/guides/logging-and-error-handling.html.md.erb (hunk 3 above)
Comment (jwoertink) [reply source/guides/logging-and-error-handling.html.md.erb:29]: Done.

File: source/guides/logging-and-error-handling.html.md.erb (hunk 2 above)
Comment (jwoertink) [reply source/guides/logging-and-error-handling.html.md.erb:152]: Not sure if I fully understood, but I added an extra example at the bottom. Let me know if that works.

Comment (paulcsmith) [reply source/guides/logging-and-error-handling.html.md.erb:152]: Sorry did a poor job explaining.

What I meant is that these line 86-111 should move down and a new section should be above it. Something roughly like this:

```markdown
## Handling Errors

Let's say you have a `MyCustomerError` class in your app and when it is raised, you want to show a custom message to your users.

You would go to `Errors::Show` in <path> and add a `handle_error`

def handle_error(e : MyCustomError)
  if html?
    render page and stuff
  else
    render son and stuff
  end
end
```

Then after this I'd show what is currently there, but mention it as more of a "this is helpful for libraries where you want to provide a default status code other than 500. But to stick to `handle_error` for most cases. Does that make sense?

Comment (jwoertink) [reply source/guides/logging-and-error-handling.html.md.erb:152]: Sorry, got a little busy there. Ok, pushed up a change. I'm feeling like this is a little redundant now, but I guess redundancy is ok in the sake of clarity. Let me know if that looks better, or if you have any other suggestions.

File: source/guides/logging-and-error-handling.html.md.erb [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+
+```crystal
+def handle_error(e : MyCustomError)
+  if html?
+    render my_custom_error_page
+  else
+    render json_or_whatever
+  end
+end
+```
```
Comment (paulcsmith) [inline source/guides/logging-and-error-handling.html.md.erb:96]: I love the new section and the note about `show_debug_output`

File: source/guides/logging-and-error-handling.html.md.erb [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+When an exception is thrown in your code, you don't want your user left with a blank page, so Lucky has a built in way of handling these errors.
+
+When in development, Lucky uses the [ExceptionPage](https://github.com/crystal-loot/exception_page) shard to display a helpful page with your stack trace, and exception message. The option to display the error page or not is in `config/error_handler.cr`. Set the `settings.show_debug_output` option to `false` if you need to see how the errors are handled in production (i.e. json response for an api).
+
+Let's say you have an error class `MyCustomError` in your app. When this error is raised, you want to show a custom error to your users. Open up the `Errors::Show` in `src/actions/errors/show.cr`, and add your `handle_error` method like this.
+
+```crystal
+def handle_error(e : MyCustomError)
+  if html?
+    render my_custom_error_page
```
Comment (paulcsmith) [inline source/guides/logging-and-error-handling.html.md.erb]: Maybe `render_error_page title: "Custom error message.", status: 418`

File: source/guides/logging-and-error-handling.html.md.erb [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+  else
+    render json_or_whatever
```
Comment (paulcsmith) [inline source/guides/logging-and-error-handling.html.md.erb]: `json({error: "Oh no!"})`?

**Review by paulcsmith — COMMENTED** [review]
Super minor comments. Looks awesome

File: source/guides/logging-and-error-handling.html.md.erb (hunk 6 above)
Comment (paulcsmith) [reply source/guides/logging-and-error-handling.html.md.erb]: Or maybe: `json Errors::ShowSerializer.new("Yikes!"), status: 418`. This matches the existing error patterns generated by `Errors::Show`

**Review by paulcsmith — APPROVED** [review]

**Pull request merged**